src/rospkg/common.py

```diff
@@ -45,13 +45,28 @@ class ResourceNotFound(Exception):
     A ROS filesystem resource was not found.
     """
 
-    def __init__(self, msg, ros_paths=None):
+    def __init__(self, msg, ros_paths=None, deps_sofar=None, deps_unavailable=None):
```

Comment:
The class `ResourceNotFound` is used in several different cases and in many of them there is no concept of dependencies. Therefore I don't think this information should be added to this class.